controllers/machinehealthcheck_controller_test.go

```diff
@@ -46,906 +45,859 @@ import (
-	providerID := fmt.Sprintf("test:////%v", uuid.NewUUID())
-	return &unstructured.Unstructured{
-		Object: map[string]interface{}{
-			"apiVersion": "infrastructure.cluster.x-k8s.io/v1alpha3",
-			"kind":       "InfrastructureMachine",
-			"metadata": map[string]interface{}{
-				"generateName": "test-mhc-machine-infra-",
-				"namespace":    machine.Namespace,
-			},
-			"spec": map[string]interface{}{
-				"providerID": providerID,
-			},
-		},
-	}, providerID
-}
-
-var _ = Describe("MachineHealthCheck", func() {
-	Context("on reconciliation", func() {
-		var (
-			cluster *clusterv1.Cluster
-			mhc     *clusterv1.MachineHealthCheck
-		)
-
-		var (
-			nodes             []*corev1.Node
-			machines          []*clusterv1.Machine
-			fakeNodesMachines = func(n int, healthy bool, nodeRef bool) {
-
-				for i := 0; i < n; i++ {
-					machine := newMachine(mhc, cluster)
-					infraMachine, providerID := newInfraMachine(&machine)
-					node := newNode()
-					if !healthy {
-						setNodeUnhealthy(&node)
-					}
-					Expect(testEnv.Create(ctx, infraMachine)).To(Succeed())
-					fmt.Fprintf(GinkgoWriter, "inframachine created: %s\n", infraMachine.GetName())
-					machine.Spec.InfrastructureRef = corev1.ObjectReference{
-						APIVersion: infraMachine.GetAPIVersion(),
-						Kind:       infraMachine.GetKind(),
-						Name:       infraMachine.GetName(),
-					}
-					infraMachinePatch := client.MergeFrom(infraMachine.DeepCopy())
-					Expect(unstructured.SetNestedField(infraMachine.Object, true, "status", "ready")).To(Succeed())
-					Expect(testEnv.Status().Patch(ctx, infraMachine, infraMachinePatch)).To(Succeed())
-
-					machineStatus := machine.Status
-					Expect(testEnv.Create(ctx, &machine)).To(Succeed())
-					fmt.Fprintf(GinkgoWriter, "machine created: %s\n", machine.GetName())
-
-					if nodeRef {
-						node.Spec.ProviderID = providerID
-						nodeStatus := node.Status
-						Expect(testEnv.Create(ctx, &node)).To(Succeed())
-						fmt.Fprintf(GinkgoWriter, "node created: %s\n", node.GetName())
-						nodePatch := client.MergeFrom(&node)
-						node.Status = nodeStatus
-						Expect(testEnv.Status().Patch(ctx, &node, nodePatch)).To(Succeed())
-						nodes = append(nodes, &node)
-
-						machinePatch := client.MergeFrom(machine.DeepCopy())
-						machine.Status = machineStatus
-						machine.Status.NodeRef = &corev1.ObjectReference{
-							Name: node.Name,
-						}
-
-						now := metav1.NewTime(time.Now())
-						machine.Status.LastUpdated = &now
-						Expect(testEnv.Status().Patch(ctx, &machine, machinePatch)).To(Succeed())
-					}
+func TestMachineHealthCheck_Reconcile(t *testing.T) {
+	t.Run("it should ensure the correct cluster-name label with no existing labels exist", func(t *testing.T) {
+		g := NewWithT(t)
+		ctx := context.TODO()
+		cluster, mhc := setup(g)
+
+		mhc.Spec.ClusterName = cluster.Name
+		mhc.Labels = map[string]string{}
+		mhc.SetGenerateName("correct-cluster-name-label-with-no-existing-labels-")
```

Comment:
being in a separated ns, might be that having a specific generated name for each test does not add much values. Should we drop?